lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1072,13 +1075,42 @@ instance NFData DecentralizationLevel
 instance Buildable DecentralizationLevel where
     build = build . unDecentralizationLevel
 
+-- | The maximum size of a serialized `TokenBundle` (`_maxValSize` in the Alonzo
+-- ledger)
+newtype TokenBundleMaxSize = TokenBundleMaxSize
+    { unTokenBundleMaxSize :: Quantity "byte" Word16 }
+    deriving (Bounded, Eq, Generic, Show)
+
+instance NFData TokenBundleMaxSize
+
+instance Arbitrary TokenBundleMaxSize where
```

Comment:
☝️ Sneaked in this non-orphan instance 🙂 Think it makes perfect sense for simple types like this.